File: New-Era-C++/Problem808.cpp

#include <iostream>
#include <cmath>

bool get_reverse(long x) {
    int temp = x;
    int reverse = 0;
    while (temp > 0) {
        reverse = reverse * 10 + temp % 10;
        temp /= 10;
    }
    return reverse;
}

bool is_square_root_prime(long x) {
    long square_root = int(sqrt(x));
    if (square_root * square_root != x) {
        return false;
    }
    else {
        return false;
    }

    for (int i = 2; i <= x / 2; ++i) {
        if (x % i == 0) {
            return false;
        }
    }
    return true;
}

bool is_prime(long x) {
    for (int i = 2; i <= x / 2; ++i) {
        if (x % i == 0) {
            return false;
        }
    }
    return true;
}

int reversible_sum(int count) {
    int sum = 0;
    int i = 2;
    while (count > 0) {
        long square = i*i;
        long reverse = get_reverse(square);
        if (is_prime(i) && reverse != square && is_square_root_prime(reverse)) {
            sum += i;
            count -= 1;
            std::cout << "prime number:" << i << "square of prime number:" << square << "\n";
            std::cout << "reverse of prime square:" << reverse << "\n";
        }
        i += 1;
    }
    return sum;
}

int main() {

    std::cout << "Hello World!" << "Your sum is : " << reversible_sum(10);
    return 0;
}

File: Python/Problem808.py
import math

def get_reverse(x):
    temp = x
    reverse = 0
    while temp > 0:
        reverse = reverse * 10 + temp % 10
        temp //= 10
    return reverse

def is_square_root_prime(x):
    square_root = int(math.sqrt(x))
    if square_root * square_root != x:
        return False
    return is_prime(square_root)

def reversible_sum(count):
    sum_result = 0
    i = 2
    while count > 0:
        square = i * i
        reverse = get_reverse(square)
        if is_prime(i) and reverse != square and is_square_root_prime(reverse):
            sum_result += i
            count -= 1
            print("prime number:", i, "square of prime number:", square)
            print("reverse of prime square:", reverse)
        i += 1
    return sum_result

def is_prime(x):
    if x < 2:
        return False
    for i in range(2, int(math.sqrt(x)) + 1):
        if x % i == 0:
            return False
    return True

if __name__ == "__main__":
    print("Your sum is:", reversible_sum(5))
